trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid.F

```diff
@@ -0,0 +1,350 @@
+! This module defines and instantiates objects
+! for a hybrid persistence levelpool type
+! reservoir. The hybrid reservoir struct
+! inherits input and output stucts from the
+! reservoir base module and calls instantiation
+! of these into sub-objects. The hybrid
+! reservoir struct also points to structs for
+! hybrid parameters and state and calls
+! instantiation of these into sub-objects.
+! A pointer to a levelpool reservoir object
+! is also held in state, and this module
+! instantiates that levelpool object. There
+! is also a subroutine to run hybrid release
+! that is derived from the reservoir base
+! struct interface to run release.
+
+! FIXME: Change value for WCOS
+#define log_warning 6
+
+module module_persistence_levelpool_hybrid
+
+    use module_persistence_levelpool_hybrid_parameters
+    use module_persistence_levelpool_hybrid_state
+    use module_levelpool
+    use module_reservoir_utilities
+    use module_reservoir_base
+    use module_read_timeslice_data
+    implicit none
+
+    ! Extend/derive hybrid struct from the abstract base
+    ! struct for reservoirs.
+    type, extends(reservoir_base_struct) :: persistence_levelpool_hybrid_struct
```

Comment:
I dont love the naming convention. 
I would remove struct. I would indicate the inheritance in the name. 
reservoir_levelpool_persistenve_hybrid
This is just my opinion.

Reply:
If we keep naming modules with the pattern module_<name>, we could drop struct from the type name without problem. I actually find naming module starting to with module and odd naming pattern but it was clearly established in the existing code.

Reply:
I would drop it, there's nothin sacred about "module_". If we can keep in mind the importan objects, we are good.

Reply:
Based on this and a few other comments, we should revive, centralize, and update our style convention guide. I see you all following some conventions that I think few of us see need to respect at NCAR. I do think they are good points for discussion and we need to centralize what we think is the best way to go.

Reply:
Can we add a linter to the pipeline?

Reply:
I would love that! I think they cost money and I'm not sure we could deploy it in CI. But still , a linter would be invaluatble. There are no free fotran linters that I know of

Reply:
Why not `type, extends(levelpool_struct) :: persistence_levelpool_hybrid_struct`?

Reply:
@jdmattern-noaa fixing. From Donald's advice, changing all 'struct' names to 'type'. We don't inherit the hybrid from levelpool because we originally designed the hybrid to consist of both levelpool and persistence objects. We decided not to have a separate persistence object though but kept the original design otherwise.